### PR TITLE
Load Subnet proxy and API key from the env if found

### PR DESCRIPTION
**Pull request by vadmeste**
## Description

Avoid sending calls to MinIO cluser to fetch for Subnet proxy
 and API key if specified in the environment

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/subnet-utils.go [hunk 1]
```diff
@@ -66,8 +66,9 @@ mr/cKCUyBL7rcAvg0zNq1vcSrUSGlAmY3SEDCu3GOKnjG/U4E7+p957ocWSV+mQU
 			Usage: "use in environments without network access to SUBNET (e.g. airgapped, firewalled, etc.)",
 		},
 		cli.StringFlag{
-			Name:  "api-key",
-			Usage: "API Key of the account on SUBNET",
+			Name:   "api-key",
+			Usage:  "API Key of the account on SUBNET",
+			EnvVar: "_MC_SUBNET_API_KEY",
```
Comment (harshavardhana) [inline cmd/subnet-utils.go:71]:
```suggestion
			EnvVar: "MC_SUBNET_API_KEY",
```

File: cmd/subnet-utils.go [hunk 2]
```diff
@@ -289,8 +290,19 @@ func setGlobalSubnetProxyFromConfig(alias string) error {
 		return nil
 	}
 
+	var (
+		proxy     string
+		supported bool
+	)
+
+	if env, ok := os.LookupEnv("_MC_SUBNET_PROXY_URL"); ok {
```
Comment (harshavardhana) [inline cmd/subnet-utils.go:298]:
```suggestion
	if env, ok := os.LookupEnv("MC_SUBNET_PROXY_URL"); ok {
```

File: cmd/support-profile.go [hunk 3]
```diff
@@ -208,8 +208,8 @@ func execSupportProfile(ctx *cli.Context, client *madmin.AdminClient, alias stri
 	if !globalAirgapped {
 		_, e := uploadFileToSubnet(alias, profileFile, reqURL, headers)
 		if e != nil {
-			failureClr.Println("\nUnable to upload profile file to SUBNET.", e.Error())
-			successClr.Printf("It has been saved locally at '%s'\n", profileFile)
+			failureClr.Println("\nUnable to upload profile file to SUBNET:", e.Error())
+			successClr.Printf("Profiling data are saved locally at '%s'\n", profileFile)
```
Comment (harshavardhana) [inline cmd/support-profile.go:212]:
```suggestion
			successClr.Printf("Saving profiling data locally at '%s'\n", profileFile)
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/subnet-utils.go [hunk 4]
```diff
@@ -289,8 +290,19 @@ func setGlobalSubnetProxyFromConfig(alias string) error {
...
+	var (
+		proxy     string
+		supported bool
+	)
+
+	if env, ok := os.LookupEnv("MC_SUBNET_PROXY_URL"); ok {
+		proxy = env
+		supported = env != ""
+	} else {
+		proxy, supported = getKeyFromSubnetConfig(alias, "proxy")
```
Comment (anjalshireesh) [inline cmd/subnet-utils.go:302]: The comment on line#305 should also move here

**Review by anjalshireesh — APPROVED** [review]
LGTM, minor nit.

File: cmd/subnet-utils.go (hunk 4 above)
Comment (harshavardhana) [inline cmd/subnet-utils.go:302]:
```suggestion
	        // get the subnet proxy config from MinIO if available
		proxy, supported = getKeyFromSubnetConfig(alias, "proxy")
```

File: cmd/subnet-utils.go [hunk 5]
```diff
@@ -289,8 +290,19 @@ func setGlobalSubnetProxyFromConfig(alias string) error {
...
+	}
+
 	// get the subnet proxy config from MinIO if available
```
Comment (harshavardhana) [inline cmd/subnet-utils.go:305]:
```suggestion
```

**Review by dormanze — APPROVED** [review]
LGTM

**Comment by harshavardhana** [comment]
@vadmeste PTAL at CI failures

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**